CarRentals/CarRentals/Controllers/CarCRUD.cs

```diff
@@ -1,80 +1,60 @@
-﻿using CarRentals.Exceptions;
+﻿using CarRentals.Interfaces;
 using Models;
 using System;
 using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 using System.Text;
 using System.Text.Json;
 using System.Text.Json.Serialization;
+using System.Configuration;
 
 namespace CarRentals
 {
-    public class CarCRUD
+    public class CarCRUD : IDataProcessing
     {
-        const string PATH = @"Cars.json";
+        string PATH = ConfigurationManager.AppSettings["JsonFile"].ToString();
 
         JsonSerializerOptions options = new()
         {
             WriteIndented = true,
             Converters = { new JsonStringEnumConverter() }
         };
 
-
         private List<Car> CarList;
 
-
         public CarCRUD()
         {
             ReadJson();
         }
 
-
-      
         public void Create(Car car)
         {
-            CarList.Add(car);
+            if(car != null)
+                CarList.Add(car);
             SaveChanges();
         }
 
-
         public Car Get(int id)
         {
-            Car Result = null;
-
-            foreach (Car car in CarList)
-            {
-                if (id == car.Id)
-                {
-                    Result = car;
-                    break;
-                }
-            }
-
-            if(Result == null)
-            {
-                throw new CarNotFoundException("The car was not found");
-            }
-
+            Car Result = CarList.Where(car => car.Id == id).FirstOrDefault();
             return Result;
         }
 
-
         public Car Update(Car car)
         {
+            Car getCar = Get(car.Id);
 
-            for (var i = 0; i < CarList.Count; i++)
+            if (getCar != null)
             {
-                if (car.Id == CarList[i].Id)
-                {
-                    CarList[i] = car;
-                }
+                CarList[CarList.IndexOf(getCar)] = car;
```

Comment:
can you think in any other way to update the car that doesn't involve using the element index?

Reply:
Thanks for your reply. Sincerily I've thought in other ways to do that but I think that it's worst.
i thought calling to the methods delete and after create to replace the item without use the index, but I believe that is worst than use it. 
I tried with LINQ, but I didn't find any statement that could help me.
However I'm going to continue reviewing this method to improve it.

Reply:
that's great! did you try just fetching the object from the list and updating it? not assigning one object to the other like `obj1 = car` (same thing you are doing now), but updating the properties? would look like something like this (pseudo code):
```csharp
var existingCar = CarList.Where(..).FirstOrDefault();
existingCar.Doors = car.Doors;
// do the same of other properties
// profit
```

we can discuss today in our meeting if this worked or not